src/ripple/app/tests/Offer.test.cpp

```diff
@@ -235,12 +253,393 @@ class Offer_test : public beast::unit_test::suite
             env.require (balance (carol, USD2 (50)));
         }
     }
+
+    void
+    testInsufficientReserve ()
+    {
+        testcase ("Insufficient Reserve");
+
+        // If an account places an offer and its balance
+        // *before* the transaction began isn't high enough
+        // to meet the reserve *after* the transaction runs,
+        // then no offer should go on the books but if the
+        // offer partially or fully crossed the tx succeeds.
+
+        using namespace jtx;
+
+        auto const gw = Account("gateway");
+        auto const USD = gw["USD"];
+
+        auto const usdOffer = USD(1000);
+        auto const xrpOffer = XRP(1000);
+
+        // No crossing:
+        {
+            Env env(*this);
+            env.fund (XRP(1000000), gw);
+
+            // The fee that's charged for transactions
+            auto const f = env.current ()->fees ().base;
+            auto const r = reserve (env, 0);
```

Comment:
There was a default parameter, why aren't you using it? e.g. `reserve(env)`. Personally, I would not use default parameters.